src/Akka.Streams.Kafka/Helpers/PartitionEventHandler.cs

```diff
@@ -69,54 +76,69 @@ public void OnAssign(IImmutableSet<TopicPartition> assignedTopicPartitions, IRes
             public void OnStop(IImmutableSet<TopicPartition> topicPartitions, IRestrictedConsumer consumer)
             {
             }
+
+            public override string ToString() => "EmptyHandler";
         }
         
         /// <summary>
         /// Handler allowing to pass custom stage callbacks.
         /// </summary>
-        internal class AsyncCallbacks : IPartitionEventHandler
+        internal sealed class AsyncCallbacks : IPartitionEventHandler
         {
+            private readonly IAutoSubscription _subscription;
+            private readonly IActorRef _sourceActorRef;
+            
             private readonly Action<IImmutableSet<TopicPartition>> _partitionAssignedCallback;
             private readonly Action<IImmutableSet<TopicPartitionOffset>> _partitionRevokedCallback;
-            private readonly Action<IImmutableSet<TopicPartitionOffset>> _partitionLostCallback;
 
-            public AsyncCallbacks(Action<IImmutableSet<TopicPartition>> partitionAssignedCallback,
-                Action<IImmutableSet<TopicPartitionOffset>> partitionRevokedCallback, 
-                Action<IImmutableSet<TopicPartitionOffset>> partitionLostCallback)
+            public AsyncCallbacks(IAutoSubscription subscription, 
+                IActorRef sourceActorRef,
+                Action<IImmutableSet<TopicPartition>> partitionAssignedCallback,
+                Action<IImmutableSet<TopicPartitionOffset>> partitionRevokedCallback)
             {
                 _partitionAssignedCallback = partitionAssignedCallback;
                 _partitionRevokedCallback = partitionRevokedCallback;
-                _partitionLostCallback = partitionLostCallback;
+                _subscription = subscription;
+                _sourceActorRef = sourceActorRef;
             }
-
-            /// <inheritdoc />
+            
             public void OnRevoke(IImmutableSet<TopicPartitionOffset> revokedTopicPartitions, IRestrictedConsumer consumer)
             {
+                _subscription.RebalanceListener
```

Comment:
Added support for the `Option<IActorRef>` rebalance listener - which is very useful for testing.